manifests/init.pp

```diff
@@ -67,18 +67,20 @@
       default: { $ilogen = 1 }
     }
     exec{"/sbin/hponcfg -f ${settingsfile} -l ${logfile}":
+
+    if $manage_package {
+      package { 'hponcfg':
+        ensure => 'present',
+        before => Exec["hponcfg -f ${settingsfile} -l ${logfile}"],
+      }
+    }
+
       onlyif      => 'test -e /sbin/hponcfg',
       path        =>'/bin:/usr/sbin:/usr/bin',
       refreshonly => true,
       subscribe   => File[$settingsfile],
       timeout     => 0,
     }
-    
-    if ! defined(Package['hponcfg']) {
```

Comment:
I think I put this here in case someone else had it define elsewhere in their puppet code.